Userland/Libraries/LibX86/Instruction.cpp

```diff
@@ -1336,6 +1338,62 @@ static String sib_to_string(u8 rm, u8 sib)
     return builder.to_string();
 }
 
+String MemoryOrRegisterReference::to_string_a64() const
+{
+    if (is_register())
+        return register_name(static_cast<RegisterIndex32>(m_register_index));
+
+    bool has_displacement = false;
+    switch (mod()) {
+    case 0b01:
+    case 0b10:
+        has_displacement = true;
+    }
+    if (m_has_sib && (m_sib & 7) == 5)
+        has_displacement = true;
+
+    String base;
+    switch (rm()) {
+    case 0:
+        base = "rax";
+        break;
+    case 1:
+        base = "rcx";
+        break;
+    case 2:
+        base = "rdx";
+        break;
+    case 3:
+        base = "rbx";
+        break;
+    case 6:
+        base = "rsi";
+        break;
+    case 7:
+        base = "rdi";
+        break;
+    case 5:
+        if (mod() == 0)
+            base = String::formatted("{:#08x}", m_displacement32);
+        else
+            base = "rbp";
+        break;
+    case 4:
+        base = sib_to_string(m_rm_byte, m_sib);
+        break;
+    }
+
+    if (!has_displacement)
+        return base;
+
+    String displacement_string;
+    if ((i32)m_displacement32 < 0)
+        displacement_string = String::formatted("-{:#x}", -(i32)m_displacement32);
+    else
+        displacement_string = String::formatted("+{:#x}", m_displacement32);
+    return String::formatted("{}{}", base, displacement_string);
```

Comment:
cant we just always cast?
String::format should have a setting to always print the sign

Reply:
`:+#x`